### ISSUE TITLE
elkai install  failed

### ISSUE THREAD

**Issue by stone7300** [pypa/packaging-problems#801]
### Problem description

 CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

**Comment by abravalheri** [comment]
To be able to help we need a complete description that explains what trouble the user is getting.
I recommend adding the error messages and stack traces to the issue text.
Adding a [minimal reproducer](https://stackoverflow.com/help/minimal-reproducible-example) is even better and in some cases fundamental...

**Comment by henryiii** [comment]
What backend (modified setuptools, scikit-build (classic), scikit-build-core) would be useful info too.

**Comment by stone7300** [comment]
pip install elkai
Collecting elkai
  Using cached elkai-2.0.1.tar.gz (227 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: elkai
  Building wheel for elkai (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for elkai (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      *** scikit-build-core 0.10.5 using CMake 3.30.2 (wheel)
      *** Configuring CMake...
      2024-08-30 09:13:28,380 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
      loading initial cache file C:\Users\ADMINI~1\AppData\Local\Temp\tmpjbshhotv\build\CMakeInit.txt
      -- Building for: NMake Makefiles
      CMake Error at CMakeLists.txt:3 (project):
        Running

         'nmake' '-?'

        failed with:

         no such file or directory


      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for elkai
Failed to build elkai
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (elkai)

**Comment by stone7300** [comment]
I used python version is  3.12.4，Does it too  new？

**Comment by stone7300** [comment]
I am waiting for your answer in line. Thank you!

**Comment by henryiii** [comment]
There are pre-built wheels for Python 3.7-3.11, so using Python 3.11 would work without trying to build the library. I believe the message you are seeing (NMake not found) is because CMake couldn't find the MSVC compiler on your system. Do you have Visual Studio installed? You need that to build code. The library you are trying to build doesn't seem to have been touched in a year, but you could try opening an issue there to get Python 3.12 (and 3.13) wheels built. The author does seem to have responded quickly to the last request for missing wheels.

Binary wheel list here: https://pypi.org/project/elkai/#files

Repo here: https://github.com/fikisipi/elkai

**Comment by stone7300** [comment]
pip install elkai
Collecting elkai
  Downloading elkai-2.0.1-cp310-cp310-win_amd64.whl (158 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.9/158.9 kB 2.4 MB/s eta 0:00:00
Installing collected packages: elkai
Successfully installed elkai-2.0.1

[notice] A new release of pip available: 22.3.1 -> 24.2
[notice] To update, run: python.exe -m pip install --upgrade pip
I downloaded another edition of python 3.10.10,
Does it mean the package has been installed successfully?

**Comment by henryiii** [comment]
Yes, "Successfully installed elkai-2.0.1" means it was successfully installed.
